src/main/java/edu/harvard/iq/dataverse/MetadataBlockServiceBean.java

```
@@ -54,29 +54,52 @@ public List<MetadataBlock> listMetadataBlocksDisplayedOnCreate(Dataverse ownerDa
         CriteriaQuery<MetadataBlock> criteriaQuery = criteriaBuilder.createQuery(MetadataBlock.class);
         Root<MetadataBlock> metadataBlockRoot = criteriaQuery.from(MetadataBlock.class);
         Join<MetadataBlock, DatasetFieldType> datasetFieldTypeJoin = metadataBlockRoot.join("datasetFieldTypes");
-        Predicate displayOnCreatePredicate = criteriaBuilder.isTrue(datasetFieldTypeJoin.get("displayOnCreate"));
-
+        
         if (ownerDataverse != null) {
             Root<Dataverse> dataverseRoot = criteriaQuery.from(Dataverse.class);
-            Join<Dataverse, DataverseFieldTypeInputLevel> datasetFieldTypeInputLevelJoin = dataverseRoot.join("dataverseFieldTypeInputLevels", JoinType.LEFT);
+            Join<Dataverse, DataverseFieldTypeInputLevel> datasetFieldTypeInputLevelJoin = 
+                dataverseRoot.join("dataverseFieldTypeInputLevels", JoinType.LEFT);
+
+            // Subquery to check if the input level exists
+            Subquery<Long> inputLevelSubquery = criteriaQuery.subquery(Long.class);
+            Root<DataverseFieldTypeInputLevel> subqueryRoot = inputLevelSubquery.from(DataverseFieldTypeInputLevel.class);
+            inputLevelSubquery.select(criteriaBuilder.literal(1L))
+                .where(
+                    criteriaBuilder.equal(subqueryRoot.get("dataverse"), dataverseRoot),
+                    criteriaBuilder.equal(subqueryRoot.get("datasetFieldType"), datasetFieldTypeJoin)
+                );
```

I suggested changing this in the earlier PR since, at that point, displayOnCreate could be false incorrectly due to the default of setting it false in the code changes above. Is this still an issue - i.e. the default should be null and this subquery should only return 1 if displayOnCreate is not null?

---

@qqmyers You raise a good point. Currently, when no input level is configured, displayOnCreate defaults to false, which might hide fields that should be shown by default. We should consider making displayOnCreate nullable in DatasetFieldType and only setting it to false when explicitly configured in the input level. This would allow the default displayOnCreate value from the field type to be used when no input level is specified. Do you agree? Or am I missing something?

---

Yes - that's what I was suggesting - I think you then have to adjust this subquery to add displayOnCreate not null so that you get the default when it is null.

---

@qqmyers Changes added in the commit 15fc52ce8ad1776f78ee0711b9a3f69c3e70a775

---

Looks good!

---

Sorry - Or maybe not - I think the nullable Boolean is needed on the DataverseFieldTypeInputLevel and not on the DatasetFieldType - the field type will always be true or false, but the input level might not set it at all.